cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql

```diff
@@ -82,30 +118,28 @@ predicate isNonConst(DataFlow::Node node) {
   //       are considered as possible non-const sources
   // The function's output must also not be const to be considered a non-const source
   exists(Function func, CallInstruction call |
-    // NOTE: could use `Call` getAnArgument() instead of `CallInstruction` but requires two
-    // variables representing the same call in ordoer to use `callOutput` below.
-    exists(Expr arg |
-      call.getPositionalArgumentOperand(_).getDef().getUnconvertedResultExpression() = arg and
-      arg = node.asDefiningArgument()
-    )
-    or
-    call.getUnconvertedResultExpression() = node.asIndirectExpr()
-  |
+    not func.hasDefinition() and
     func = call.getStaticCallTarget() and
+    isInsideSourceRoot(func)
```

Comment:
I agree. The current set of sources is ridiculously large, though. For example, without this change every single external function in https://github.com/mirror/libX11/blob/master/include/X11/Xlib.h would be marked as a source on `ImageMagick` - which is why it doesn't terminate on Schack's PR.

With this change, only external that's controlled by the developer will be marked as sources. So if I have a header with something like:
```cpp
// MyCompanyExternalFunctions.h
extern Foo* getFoo();
```
then this will still be a source in the query.

Reply:
This is no longer needed because of the changes in 61597f5ac79b2a38285a2299fc0d5439d6257ae0 🎉